### PR TITLE
fix(sniplet): #RBS-122 move min max delay warning

### PR DESCRIPTION
**Pull request by alicedraillard**
## Describe your changes
fixes following the testing of RBS-122:
- ADML user can book without regard of the delays
- days are counted as full
- fix timezone problem

## Checklist tests
with adml or super admin account: try to create a booking outside the delay boudaries
with normal account: test the delay boundaries (eg if the boundary is 16:00 try to have a start date at 15:59 or 16:01)

## Issue ticket number and link
RBS-122 fix https://entsupport.gdapublic.fr/browse/RBS-122

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -24,36 +24,36 @@ clean () {
 
 buildNode () {
   #jenkins
-  echo "[buildNode] Get branch name from jenkins env..."
```
Comment (alicedraillard) [inline build.sh]: used to launch ng2, will be deleted before merge

File: src/main/resources/public/template/behaviours/sniplet-calendar-rbs-booking.html [hunk 2]
```diff
@@ -12,16 +12,6 @@ <h2 class="backgroundBlueTitle" ng-if="vm.canEditEvent">
         <div ng-if="vm.recurrenceOrMultidaysAdded()" class="info">
             <i18n>rbs.calendar.sniplet.recurrence.multiday.edition.warning</i18n>
         </div>
-        <!-- Min delay not respected warning -->
```
Comment (alicedraillard) [inline src/main/resources/public/template/behaviours/sniplet-calendar-rbs-booking.html:15]: move these warnings below

File: src/main/resources/public/ts/sniplets/calendar-rbs-booking.sniplet.ts [hunk 3]
```diff
@@ -63,6 +63,8 @@ interface IViewModel {
 
     userIsAdml(): boolean;
 
+    userIsSuperAdmin(): boolean;
```
Comment (alicedraillard) [inline src/main/resources/public/ts/sniplets/calendar-rbs-booking.sniplet.ts:66]: add super admin condition to match RBS behaviour

File: src/main/resources/public/ts/sniplets/calendar-rbs-booking.sniplet.ts [hunk 4]
```diff
@@ -522,14 +531,20 @@ class ViewModel implements IViewModel {
      * Returns true if the minimum delay is compliant
      */
     checkMinDelay(): boolean {
-        return BookingDelayUtil.checkMinDelay(this.editedBooking);
+        let delaybooking = angular.copy(this.editedBooking);
```
Comment (alicedraillardcgi) [inline src/main/resources/public/ts/sniplets/calendar-rbs-booking.sniplet.ts:534]: handle date offset

File: src/main/resources/public/ts/sniplets/calendar-rbs-booking.sniplet.ts [hunk 5]
```diff
@@ -522,14 +531,20 @@ class ViewModel implements IViewModel {
...
+        this.prepareBookingStartAndEnd(delaybooking);
```
Comment (alicedraillard) [inline src/main/resources/public/ts/sniplets/calendar-rbs-booking.sniplet.ts:535]: handle date offset

File: src/main/resources/public/ts/sniplets/calendar-rbs-booking.sniplet.ts [hunk 6]
```diff
@@ -522,14 +531,20 @@ class ViewModel implements IViewModel {
...
+
+        return (BookingDelayUtil.checkMinDelay(delaybooking) || this.userIsAdml() || this.userIsSuperAdmin());
```
Comment (alicedraillard) [inline src/main/resources/public/ts/sniplets/calendar-rbs-booking.sniplet.ts:537]: add super admin and admin condition

File: src/main/resources/public/ts/sniplets/calendar-rbs-booking.sniplet.ts [hunk 7]
```diff
@@ -522,14 +531,20 @@ class ViewModel implements IViewModel {
...
     }
 
     /**
      * Returns true if the maximum delay is compliant
      */
     checkMaxDelay(): boolean {
-       return BookingDelayUtil.checkMaxDelay(this.editedBooking);
+        let delaybooking = angular.copy(this.editedBooking);
```
Comment (alicedraillard) [inline src/main/resources/public/ts/sniplets/calendar-rbs-booking.sniplet.ts]: same 2 behaviours added

File: src/main/resources/public/ts/sniplets/calendar-rbs-booking.sniplet.ts (hunk 4 above)
Comment (BastienBenard) [inline src/main/resources/public/ts/sniplets/calendar-rbs-booking.sniplet.ts]: typage possible ?

File: src/main/resources/public/ts/sniplets/calendar-rbs-booking.sniplet.ts (hunk 7 above)
Comment (BastienBenard) [reply src/main/resources/public/ts/sniplets/calendar-rbs-booking.sniplet.ts]: typage possible ?

**Review by BastienBenard — APPROVED** [review]

**Review by JinDouang — APPROVED** [review]

**Review by flmartineau — APPROVED** [review]

**Pull request merged**